control-plane-operator/controllers/hostedcontrolplane/testdata/kube-controller-manager/zz_fixture_TestControlPlaneComponents_component.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: hypershift.openshift.io/v1beta1
+kind: ControlPlaneComponent
+metadata:
+  creationTimestamp: null
+  name: kube-controller-manager
+  namespace: hcp-namespace
+  resourceVersion: "1"
+spec: {}
+status:
+  conditions:
+  - lastTransitionTime: null
+    message: kube-controller-manager Deployment Available condition not found
```

Comment:
tangential: is there a place where we are unit testing reconcileComponentStatus for cases where conditions actually exists?

Reply:
Also is there anything preventing us from testing all assets against fixtures? not only Deployment, StatefulSet

Reply:
> is there a place where we are unit testing reconcileComponentStatus for cases where conditions actually exists?

we don't, but yeah that makes sense, I will create a ticket for that.

Reply:
> Also is there anything preventing us from testing all assets against fixtures? not only Deployment, StatefulSet

not really, its just that the generic unit test, doesn't know what resources each component is generating. so we would probably need to list every possible resource and see if one exists and then create a fixture for it. I don't know if there is an easier way to do that.